drivers/uktty/ns16550/ns16550.c

```diff
@@ -129,6 +149,22 @@ static int init_ns16550(void)
 	ns16550_reg_write(NS16550_FCR_OFFSET,
 			  ns16550_reg_read(NS16550_FCR_OFFSET) &
 					 ~(NS16550_FCR_FIFO_EN));
+
+	/* Set line control parameters (8n1) */
+	lcr_mask = NS16550_LCR_WL | NS16550_LCR_STOP | NS16550_LCR_PARITY;
+	ns16550_reg_write(NS16550_LCR_OFFSET,
+			  ns16550_reg_read(NS16550_LCR_OFFSET) & ~(lcr_mask));
+	ns16550_reg_write(NS16550_LCR_OFFSET, NS16550_LCR_8N1);
+
+	/* Set DLAB to access DLL / DLM */
+	ns16550_reg_write(NS16550_LCR_OFFSET,
+			  ns16550_reg_read(NS16550_LCR_OFFSET) &
```

Comment:
I may be missing something but why do we read this register twice and write three times? Are you sure this couldn't have been shortened to a read and a write? This also reminds me of the compile-time barriers missing.

Reply:
The value of the `LCR.DLAB` affects how registers are interpreted [1], so I would keep the operations that update DLAB separate from those that update the rest of the LCR bits. For the other two writes that set the line control you're right, I'll merge them into one.

[1] https://media.digikey.com/pdf/Data%20Sheets/Texas%20Instruments%20PDFs/PC16550D.pdf